### PR TITLE
Implement String.prototype.localeCompare function

### PR DESCRIPTION
**Pull request by bzsolt**
JerryScript-DCO-1.0-Signed-off-by: Zsolt Borbély zsborbely.u-szeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp [hunk 1]
```diff
@@ -286,7 +286,48 @@ static ecma_completion_value_t
...
+                  ecma_op_to_string (this_arg),
+                  ret_value);
+
+  /* 3. */
+  ECMA_TRY_CATCH (arg_to_string_val,
+                  ecma_op_to_string (arg),
+                  ret_value);
+
+  ecma_string_t *this_string_p = ecma_get_string_from_completion_value (this_to_string_val);
+  ecma_string_t *arg_string_p = ecma_get_string_from_completion_value (arg_to_string_val);
```
Comment (galpeter) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp]: The `this_to_string_val` and `arg_to_string_val` are not completion values. They are `ecma_value_t`-s

Comment (bzsolt) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp]: Oops, you're right, I've fixed it.
Thanks.

**Comment by zherczeg** [comment]
looks good to me

**Comment by ruben-ayrapetyan** [comment]
Looks good to me

**Comment by egavrin** [comment]
Good to me

**Comment by egavrin** [comment]
Merged at f625473f9a6253c9b06f8a86ae3688d6f7920ed8

**Pull request closed without merging**